### PR TITLE
expand data handling

### PR DESCRIPTION
**Pull request by bruno-garcia**
* Being more explicit about our SDK guidelines for handling sensitive data
* Point out that changes to Relay have to happen to influence server side PII stripping
* Add to Philosophy
* Reorder pages to become a more sequential flow for learning

### CODE REVIEW

File: src/docs/sdk/data-handling.mdx [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
 
 Data handling is the standardized context in how we want SDKs help users filter data.
 
 ## Sensitive Data
 
-In older SDKs you might sometimes see elaborate constructs to allow the user to strip away sensitive data. Newer SDKs no longer have this feature as it turned out to be too hard to maintain per-SDK. Instead, only two simple config options are left:
+SDKs should not include PII or other sensitive data in the payload by default.
 
-- [_send-default-pii_](https://docs.sentry.io/error-reporting/configuration/#send-default-pii) is **disabled by default**, meaning that data that is naturally sensitive is not sent by default. That means, for example:
+- [_send-default-pii_](https://docs.sentry.io/platforms/python/configuration/options/#send-default-pii) is **disabled by default**, meaning that data that is naturally sensitive is not sent by default. That means, for example:
```
Comment (dcramer) [inline src/docs/sdk/data-handling.mdx:12]: its probably worth clarifying what this setting is here (the intent of it)

File: src/docs/sdk/data-handling.mdx [hunk 2]
```diff
@@ -1,23 +1,44 @@
...
 
   - When attaching HTTP requests to events, "raw" bodies (bodies which cannot be parsed as JSON or formdata) are removed, and known sensitive headers such as `Authorization` or `Cookies` are removed too.
 
   - User-specific information (e.g. the current user ID according to the used webframework) is not sent at all.
 
-  - Note that if a user explicitly sets a request on the scope, nothing is stripped from that request. The above rules only apply to integrations that come with the SDK.
+  - On desktop applications, the username logged in the device is not included.
 
-- [_before-send_](https://docs.sentry.io/error-reporting/configuration/#before-send) can be used to register a callback with custom logic to remove sensitive data.
+  - SDKs don't send `{{auto}}` as `user.ip`. This instructs the server to keep the connection's IP address.
```
Comment (dcramer) [inline src/docs/sdk/data-handling.mdx:20]: May be worth clarifying that client SDKs will always have an IP captured as they directly connect to the Sentry service to submit data. And also if we note this, just put an analogy that it is similar to how server-side logs work, and the data is captured for auditibility.

File: src/docs/sdk/data-handling.mdx [hunk 3]
```diff
@@ -1,23 +1,44 @@
...
+
+*Note* that if a user explicitly sets a request on the scope, nothing is stripped from that request. The above rules only apply to integrations that come with the SDK.
+
+Before sending events to Sentry, the SDKs should invokes callbacks. That allows users to remove any sensitive data client-side.
+
+- `before-send` and `event-processors` can be used to register a callback with custom logic to remove sensitive data.
```
Comment (dcramer) [inline src/docs/sdk/data-handling.mdx:26]: links on these?

File: src/docs/sdk/data-handling.mdx [hunk 4]
```diff
@@ -1,23 +1,44 @@
...
+
+### Auto instrumentation
```
Comment (dcramer) [inline src/docs/sdk/data-handling.mdx:28]: Maybe just call this Local Application State ?

File: src/docs/sdk/philosophy.mdx [hunk 5]
```diff
@@ -46,3 +46,9 @@ At no point should a customer upgrade the library and find themselves in a situa
 ## Enable Customers
 
 While we generally should try to keep the API surfaces of SDKs reasonable small. At the same time we also need to make sure that we enable customers to achieve their goals. Think about cases that the SDK maybe wouldn't be able to solve out of the box. If there are enough APIs that customers can use our SDKs in more creative ways, we generally see this as a added benefit.
+
+## Handle PII and Sensitive Data with Care
+
+App state is key to help reproduce issues. But make sure that auto instrumentations doesn't attach PII without an explicit opt-in from the user.
```
Comment (dcramer) [inline src/docs/sdk/philosophy.mdx:52]: idk how to suggest changes, but "key to helping"

**Review by dcramer — COMMENTED** [review]
prob some general language cleanup that could happen here too - maybe @imatwawana can help there

File: src/docs/sdk/data-handling.mdx (hunk 2 above)
Comment (bruno-garcia) [reply src/docs/sdk/data-handling.mdx:20]: Great point about the server-side logging, I'll add.
Wrt Sentry already capturing, AFAIK this is only true for iOS and JavaScript events (based on `event.platform=javascript` or `cocoa`. Every other platform requires `user.ip={{auto}}`

File: src/docs/sdk/data-handling.mdx (hunk 3 above)
Comment (bruno-garcia) [reply src/docs/sdk/data-handling.mdx:26]: We don't have generic docs (I'll use JS or Python as I've done in the other examples).
Actually, I'll link within the develop docs

**Comment by bruno-garcia** [comment]
Applied the updates, appreciate another stab at reviews please

**Review by AbhiPrasad — APPROVED** [review]
I think this is a good start. Do we want to add a note about PII vs. potentially sensitive data?

**Comment by bruno-garcia** [comment]
> I think this is a good start. Do we want to add a note about PII vs. potentially sensitive data?

Feel free to open a follow up, lets merge this for now as it improves things a bit

**Pull request merged**